src/Functions/FunctionsHashing.h

```diff
@@ -1053,6 +1056,10 @@ class FunctionAnyHash : public IFunction
 {
 public:
     static constexpr auto name = Impl::name;
+    static constexpr int null_magic_number = 42;
```

Comment:
I don't like that. Not fundamental enough.
Zero is better.

Reply:
But, it allow to distinguish Zero value and Nulls, which are really common default values for data types.

Kinda make sense to have it different from zero for nulls.